lib/util.js

```diff
@@ -335,6 +335,25 @@ module.exports.normalizePath = function(oldPath, relativeTo) {
   return normalized;
 }
 
+module.exports.getRemoteIframeTemplate = function() {
+  let defaultRemoteAppTemplate = fs.readFileSync(path.join(__dirname,'default-remote-app-template.html'),"utf-8");
+  return (templatePath) => {
+    let remoteAppTemplate = defaultRemoteAppTemplate;
+    if(templatePath) {
+      remoteAppTemplate = fs.readFileSync(templatePath, "utf-8");
```

Comment:
this must be changed to a promise which uses `readFile` instead of `readFileSync` since this is happening during a network call, it freezes the server while loading the file.
For performance, you may also consider putting the result into a map such that you can get from the map for every time after the first time.
Doing both will greatly improve performance when there are many clients.

Reply:
removed this logic, as not testing this path right now, so just implementing default case for now. Thanks!